tasks/main.yml

```diff
@@ -26,5 +26,8 @@
     src: etc/apt/preferences.d/deb_nodesource_com_node.pref.2
     dest: /etc/apt/preferences.d/deb_nodesource_com_node.pref
 
+- name: Uninstall existing installations of Node.js
+  apt: name=nodejs state=absent
+
```

Comment:
Since posting this issue I've created my own Ansible implementation of NVM and I no longer use nodesource.

If you were to do so, you should check for an existing install, register a variable with the currently installed version and provide some logic to determine if it should be uninstalled first.